src/main/java/bio/terra/cli/service/AxonServerService.java

```diff
@@ -0,0 +1,205 @@
+   */
+  public ClusterMetadata getClusterMetadata(UUID workspaceId, UUID resourceId) {
+    GcpResourceApi gcpResourceApi = new GcpResourceApi(apiClient);
+    return callWithRetries(
+        () -> gcpResourceApi.getDataprocClusterMetadata(workspaceId, resourceId),
+        "Failed to retrieve cluster metadata");
+  }
+
+  /**
+   * Call the Axon Server GET
+   * /api/workspaces/v1/{workspaceId}/resources/{resourceId}/gcp/dataproccluster/componentUrl:
+   *
+   * @param workspaceId the workspace that contains the resource
+   * @param resourceId the cluster resource id
+   * @param componentKey the component key, e.g. "JupyterLab"
+   */
+  public bio.terra.axonserver.model.Url getClusterComponentUrl(
+      UUID workspaceId, UUID resourceId, String componentKey) {
+    GcpResourceApi gcpResourceApi = new GcpResourceApi(apiClient);
+    return callWithRetries(
+        () -> gcpResourceApi.getDataprocClusterComponentUrl(workspaceId, resourceId, componentKey),
+        "Failed to retrieve cluster component url");
+  }
+
+  /**
+   * Call the Axon Server GET
+   * /api/workspaces/v1/{workspaceId}/resources/{resourceId}/gcp/dataproccluster/status:
+   *
+   * @param workspaceId the workspace that contains the resource
+   * @param resourceId the cluster resource id
+   */
+  public ClusterStatus getClusterStatus(UUID workspaceId, UUID resourceId) {
+    GcpResourceApi gcpResourceApi = new GcpResourceApi(apiClient);
+    return callWithRetries(
+        () -> gcpResourceApi.getDataprocClusterStatus(workspaceId, resourceId),
+        "Failed to retrieve cluster status");
+  }
+
+  /**
+   * Call the Axon Server GET
+   * /api/workspaces/v1/{workspaceId}/resources/{resourceId}/gcp/dataproccluster/start:
+   *
+   * @param workspaceId the workspace that contains the resource
+   * @param resourceId the cluster resource id
+   */
+  public void startCluster(UUID workspaceId, UUID resourceId) {
+    GcpResourceApi gcpResourceApi = new GcpResourceApi(apiClient);
+    callWithRetries(
+        () -> gcpResourceApi.putDataprocClusterStart(workspaceId, resourceId, /* wait=*/ false),
+        "Failed to start cluster");
+  }
+
+  /**
+   * Call the Axon Server GET
+   * /api/workspaces/v1/{workspaceId}/resources/{resourceId}/gcp/dataproccluster/stop:
+   *
+   * @param workspaceId the workspace that contains the resource
+   * @param resourceId the cluster resource id
+   */
+  public void stopCluster(UUID workspaceId, UUID resourceId) {
+    GcpResourceApi gcpResourceApi = new GcpResourceApi(apiClient);
+    callWithRetries(
+        () -> gcpResourceApi.putDataprocClusterStop(workspaceId, resourceId, /* wait=*/ false),
+        "Failed to stop cluster");
+  }
+
+  /**
+   * Utility method that checks if an exception thrown by the Axon Server client is retryable.
+   *
+   * @param ex exception to test
+   * @return true if the exception is retryable
+   */
+  private static boolean isRetryable(Exception ex) {
```

Comment:
outside the scope of this pr - functions following this are similar across services and can be shared code. 
could you please add a backlog ticket under 
https://verily.atlassian.net/browse/BENCH-411